QualityControl/public/object/objectTreePage.js

```diff
@@ -192,66 +192,93 @@ const treeRows = (model) => !model.object.tree ?
+      open,
+      name,
+      () => tree.toggle(),
+      {
+        paddingLeft: `${level}em`,
+      },
+    );
+    rows.push(branch);
   }
+
+  return [...rows, ...childRow];
 }
 
 /**
  * Creates a row containing specific visuals for leaf object and on selection
  * it will plot the object with JSRoot
- * @param {string} path - full name of the object
- * @param {Action} selectItem - action for plotting the object
- * @param {string} className - name of the row class
- * @param {number} padding - space needed to be displayed so that leaf is within its parent
- * @param {string} leafName - name of the object
+ * @param {string} key - An unique identifier for this branch row element (table row)
+ * @param {string} leafName - The name of this tree object element
+ * @param {() => void} onClick - The action (callback) to perform upon clicking this branch row element (table row)
+ * @param {string} className - Optional CSS class name(s) for the outer branch row element (table row)
+ * @param {object} styling - Optional CSS styling for the inner branch row element (table data)
  * @returns {vnode} - virtual node element
  */
-const leafRow = (path, selectItem, className, padding, leafName) =>
+const leafRow = (key, leafName, onClick, className = '', styling = {}) =>
   h('tr.object-selectable', {
-    key: path, title: path, onclick: selectItem, class: className, id: path,
+    key: key,
+    id: key,
+    title: leafName,
+    onclick: onClick,
+    class: className,
   }, [
-    h('td.highlight', [
-      h('span', { style: { paddingLeft: padding } }, iconBarChart()),
-      ' ',
+    h('td.highlight.flex-row.items-center.g1', {
+      style: styling,
+    }, [
+      iconBarChart(),
       leafName,
     ]),
   ]);
 
 /**
  * Creates a row containing specific visuals for branch object and on selection
  * it will open its children
- * @param {string} path - full name of the object
- * @param {ObjectTree} tree - current selected tree
- * @param {number} padding - space needed to be displayed so that branch is within its parent
+ * @param {string} key - An unique identifier for this branch row element (table row)
+ * @param {boolean} isTreeOpen - Whether the current tree object is open (expanded)
+ * @param {string} branchName - The name of this tree object element
+ * @param {() => void} onClick - The action (callback) to perform upon clicking this branch row element (table row)
+ * @param {object} styling - Optional CSS styling for the inner branch row element (table data)
  * @returns {vnode} - virtual node element
  */
-const branchRow = (path, tree, padding) =>
-  h('tr.object-selectable', { key: path, title: path, onclick: () => tree.toggle() }, [
-    h('td.highlight', [
-      h('span', { style: { paddingLeft: padding } }, tree.open ? iconCaretBottom() : iconCaretRight()),
-      ' ',
-      tree.name,
+const branchRow = (key, isTreeOpen, branchName, onClick, styling = {}) =>
+  h('tr.object-selectable', {
+    key: key,
```

Comment:
No need to specify the variable value if it has the same name as the key of the object.

So in your case, you can simply put:
```json
{
  key,
  id: key,
  title: branchName,
  onclick  // (of course you should adapt the name in method declaration as it is case sensitive)
}